### PR TITLE
Git: inlineCallback updates

### PR DESCRIPTION
**Pull request by jaredgrubb**
Two things here:
 * move a couple functions to inlineCallbacks
 * remove "fetchcount", which is not used at all in this file

This should be a non-functional change, and passes tests, pylint, pep8. Will backport to eight as well.

### CODE REVIEW

File: master/buildbot/steps/source/git.py [hunk 1]
```diff
@@ -587,34 +584,31 @@ def applyAlready(res):
...
 
-            @d.addCallback
-            def checkWithListdir(_):
-                if 'files' not in cmd.updates:
-                    # no files - directory doesn't exist
-                    return "clone"
-                files = cmd.updates['files'][0]
-                if '.git' in files:
-                    return "update"
-                elif len(files) > 0:
```
Comment (sa2ajj) [inline master/buildbot/steps/source/git.py]: I like that this is removed :)

File: master/buildbot/steps/source/git.py [hunk 2]
```diff
@@ -587,34 +584,31 @@ def applyAlready(res):
...
+        cmd.useLog(self.stdio_log, False)
+        yield self.runCommand(cmd)
+
+        if 'files' not in cmd.updates:
+            # no files - directory doesn't exist
+            defer.returnValue("clone")
+        files = cmd.updates['files'][0]
+        if '.git' in files:
+            defer.returnValue("update")
+        elif len(files) > 0:
```
Comment (sa2ajj) [inline master/buildbot/steps/source/git.py]: However I do not like that this is added :)

```
elif files:
```

should be enough, no?

Comment (jaredgrubb) [inline master/buildbot/steps/source/git.py]: I take no credit for what's there, it's actually just de-dented, the diff isnt showing it well... but yeh that could certainly just be "elif files"

Comment (sa2ajj) [inline master/buildbot/steps/source/git.py]: I understand that, though I see no reason to take a credit for improving the code :)

Comment (jaredgrubb) [inline master/buildbot/steps/source/git.py]: Updated :)

**Comment by sa2ajj** [comment]
LGTM

**Comment by tardyp** [comment]
:+1:

**Pull request merged**